src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Locale;
+import java.time.format.DateTimeParseException;
+
+public class DukeException extends Exception{
+    /**
+     * Constructor for DukeException
+     * @param msg error message
+     */
+    public DukeException(String msg) {
+        super(msg);
+    }
+
+    /**
+     * Validates the todo input is not empty
+     * @param str user input for todo task
+     * @throws DukeException if the input is empty after todo
+     */
+    static void validateToDo(String str) throws DukeException {
+        if (str.split(" ").length < 2) {
+            throw new DukeException("duke.ToDo duke.Task Missing!");
+        }
+    }
+
+    /**
+     * Validates the instruction is in the list of valid instructions
+     * @param str user input
+     * @throws DukeException if the instruction is not in the list
+     */
+    public static void validateInstn(String str) throws DukeException {
+        String[] instnArr = {"todo", "deadline", "event", "mark", "unmark", "list", "delete", "find"};
+        String instn = str.split(" ")[0].toLowerCase();
+        if (!Arrays.asList(instnArr).contains(instn)) {
+            throw new DukeException("Invalid instruction for PeWPeWPeW:(((");
+        }
+    }
+
+    /**
+     * Validates if the list index to be deleted is valid
+     * @param index list index to be deleted
+     * @param task_arr list of tasks
+     * @throws DukeException if the index is invalid
+     */
+    static void validateArrIndex(int index, ArrayList<Task> task_arr) throws DukeException {
```

Comment:
Perhaps `task_arr` can be in camelCase? I noticed the same issue in several other places too.